iree/hal/local/loaders/vmvx_module_loader.c

```diff
@@ -119,6 +113,23 @@ static iree_status_t iree_hal_vmvx_executable_create(
     }
   }
 
+  // Query the optional local workgroup size from each entry point.
+  if (iree_status_is_ok(status)) {
+    // TODO(benvanik): pack this more efficiently; this requires a lot of
+    // queries and instead could be a single packed table we can directly
+    // reference from the module. Module-level reflection attrs would help.
+    for (iree_host_size_t i = 0; i < executable->entry_fn_count; ++i) {
+      iree_string_view_t local_memory_str = iree_vm_function_reflection_attr(
+          &executable->entry_fns[i], iree_make_cstring_view("local_memory"));
+      uint32_t local_memory_size = 0;
+      if (!iree_string_view_is_empty(local_memory_str)) {
+        iree_string_view_atoi_uint32(local_memory_str, &local_memory_size);
+      }
```

Comment:
Agreed with the spirit of the TODO... but I'd expect there to be a simpler way to get this sort of int value from the compiler to the runtime than through a string :P

Does this show up in traces (how much time is `iree_hal_vmvx_executable_create` taking before/after this change)?

At least this is only running on executable creation, and not on each dispatch.

Reply:
The reflection attrs are just byte buffers in the file format so we could pack them as uint16_t's, but then tooling gets trickier as we can't easily inspect the files by dumping the flatbuffers without also specifying the type to decode as. It's fine for now as the numbers are small and are quite dense ("16" - 3 bytes with NUL - is a common value of 64KB). It's mostly about having to query each one that sucks. But this is as you say load-time only and just some extra function calls per entry point. It bothers me, but not enough to make it more complex just yet :)